### PR TITLE
Add BDD.BitSet

### PR DESCRIPTION
**Pull request by blegat**
I've started with a set of `Int` to start simple, the next step is the set of `NTuple{N, Int}`

### CODE REVIEW

File: src/BDD/BDD.jl [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+function phase_rem(variables::Vector{Ptr{CUDD.DdNode}}, x::Int)
+    phase = zeros(Cint, length(variables))
+    for i in eachindex(phase)
+        if !iszero(x & 1)
+            phase[i] = one(Cint)
+        end
+        x >>= 1
+    end
+    return phase, x
+end
```
Comment (guberger) [inline src/BDD/BDD.jl:21]: Here `variables` is used only for its length, which is computed twice. Put length as argument?

File: src/BDD/BitSet.jl [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    n = length(set.variables)
+    Base.showarg(io, set, true)
+    print(io, " with ", n, " bit")
+    if !isone(n)
+        print(io, "s")
+    end
+end
+
+Base.eltype(::Type{BitSet}) = Int
+Base.empty(::BitSet, ::Type{Int}=Int) = BitSet()
```
Comment (guberger) [inline src/BDD/BitSet.jl:27]: Pour info, ça veut dire quoi la notation `::Type{Int}=Int`, et pourquoi le `Type{}`? :-)

Comment (guberger) [reply src/BDD/BitSet.jl:27]: Ah j'ai trouvé pour la 1ere partie: le `Int` c'est parce que c'est un argument optionnel

File: src/BDD/BitSet.jl [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    end
+    return phase
+end
+minterm!(set::BitSet, x::Int) = cube(set.manager, set.variables, phase!(set, x))
+function Base.push!(set::BitSet, x::Int)
+    # `minterm!` modifies `set.root` so we need to call it before the `bddOr` line where we access `set.root`.
+    minterm = minterm!(set, x)
+    set.root = CUDD.Cudd_bddOr(set.manager, set.root, minterm)
+    return set
+end
```
Comment (guberger) [inline src/BDD/BitSet.jl:70]: We could add a function `delete!`

File: src/BDD/BitSet.jl [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    end
+end
+
+function phase!(set::BitSet, x::Int)
+    phase, x = phase_rem(set.variables, x)
+    while x > 0
+        push!(phase, iszero(x & 1) ? zero(Cint) : one(Cint))
+        # As the `manager` is only used by this struct, `bddIthVar` should be
+        # the same as `bddAddVar`.
+        var = CUDD.Cudd_bddIthVar(set.manager, Cint(length(set.variables)))
```
Comment (guberger) [inline src/BDD/BitSet.jl:57]: Why not `Cudd_bddNewVar(set.manager)`?

File: src/BDD/BDD.jl (hunk 1 above)
Comment (guberger) [reply src/BDD/BDD.jl:21]: Also, I assume there is a reason to use an array of Cint, instead of a BitArray?

File: src/BDD/BitSet.jl (hunk 4 above)
Comment (blegat) [reply src/BDD/BitSet.jl:57]: So that it works even if the user plays with the internal manager :-P

File: src/BDD/BDD.jl (hunk 1 above)
Comment (blegat) [reply src/BDD/BDD.jl:21]: Yes, `CUDD` needs a vector of `Cint`

File: src/BDD/BitSet.jl [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+function phase!(set::BitSet, x::Int)
+    phase, x = phase_rem(x, set)
+    while x > 0
+        push!(phase, iszero(x & 1) ? zero(Cint) : one(Cint))
+        # As the `manager` is only used by this struct, `bddIthVar` should be
+        # the same as `bddAddVar`.
+        var = CUDD.Cudd_bddIthVar(set.manager, Cint(length(set.variables)))
+        push!(set.variables, var)
+        set.root = CUDD.Cudd_bddAnd(set.manager, set.root, cube(set.manager, [var], [zero(Cint)]))
+        x >>= 1
```
Comment (guberger) [inline src/BDD/BitSet.jl:62]: For the efficiency, hope the user will push only with positive integers ;-)

File: src/BDD/BitSet.jl (hunk 3 above)
Comment (blegat) [reply src/BDD/BitSet.jl:72]: Good, point, I leave it for future work

**Pull request merged**